cuml/forest_inference_demo.ipynb

```diff
@@ -0,0 +1,255 @@
+    "                                    y_validation,\n",
+    "                                    num_rounds, model_path):\n",
+    "    # set the xgboost model parameters\n",
+    "    params = {'silent': 1, 'eval_metric':'error',\n",
+    "              'objective':'binary:logistic',\n",
+    "              'max_depth': 25}\n",
+    "    dtrain = xgb.DMatrix(X_train, label=y_train)\n",
+    "    # train the xgboost model\n",
+    "    bst = xgb.train(params, dtrain, num_rounds)\n",
+    "\n",
+    "    # save the trained xgboost model\n",
+    "    bst.save_model(model_path)\n",
+    "\n",
+    "    # predict the xgboost model\n",
+    "    dvalidation = xgb.DMatrix(X_validation, label=y_validation)\n",
+    "    xgb_preds = bst.predict(dvalidation)\n",
+    "\n",
+    "    # convert the predicted values from xgboost into class labels\n",
+    "    xgb_preds = np.around(xgb_preds)\n",
+    "\n",
+    "    return xgb_preds, model_path"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### setting the required parameters"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "n_rows = 10000\n",
+    "n_columns = 100\n",
+    "n_categories = 2\n",
+    "random_state = np.random.RandomState(43210)\n",
+    "\n",
+    "# enter path to the directory where the trained model will be saved\n",
+    "model_path = 'xgb.model'\n",
+    "# num of iterations for which the model is trained\n",
+    "num_rounds = 15"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### create the dataset and call the function to train and predict the xgboost model"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "X, y = make_classification(n_samples=n_rows,\n",
+    "                           n_features=n_columns,\n",
+    "                           n_informative=int(n_columns/5),\n",
+    "                           n_classes=n_categories,\n",
+    "                           random_state=random_state)\n",
+    "n_rows, n_columns = X.shape\n",
+    "train_size = 0.8\n",
+    "# convert the dataset to np.float32\n",
+    "X = X.astype(np.float32)\n",
+    "y = y.astype(np.float32)\n",
+    "# split the dataset into training and validation splits\n",
+    "X_train, X_validation, y_train, y_validation = train_test_split(\n",
+    "    X, y, train_size=train_size)\n",
+    "trained_model_preds, model_path = \\\n",
+    "    train_and_predict_xgboost_model(X_train, y_train,\n",
+    "                                    X_validation,\n",
+    "                                    y_validation,\n",
+    "                                    num_rounds, model_path)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### The load function of the ForestInference class accepts the following parameters:\n",
```

Comment:
Use 2nd level header (`##`) and simplify to something like 

```
## Load Forest Inference Library

The load function accepts the following parameters:
- `filename: str `

etc...

```

Separate Predict into it's own cell with a header:

```
## Predict classes in loaded model
```